src/translate.rs

```diff
@@ -245,12 +245,23 @@ pub fn header_items(ctxt: &rustast::ExtCtxt) -> Vec<rustast::P<rustast::Item>> {
 
 
 fn compile_rule(ctxt: &rustast::ExtCtxt, rule: &Rule) -> rustast::P<rustast::Item> {
+	let ref rule_name = rule.name;
 	let name = rustast::str_to_ident(&format!("parse_{}", rule.name));
 	let ret = rustast::parse_type(&rule.ret_type);
 	let body = compile_expr(ctxt, &*rule.expr, (&rule.ret_type as &str) != "()");
 	(quote_item!(ctxt,
 		fn $name<'input>(input: &'input str, state: &mut ParseState, pos: usize) -> RuleResult<$ret> {
-			$body
+			if cfg!(feature = "peg_trace") {
+				println!("[PEG_TRACE] Entering rule {}", $rule_name);
+                        }
+			let mut __peg_closure = move || {
```

Comment:
What's the reason for wrapping it in a closure? Does `let __peg_result = $body;` not work? I'd like to make sure that there's no performance impact when the feature is not enabled.